library-front/src/components/BookForm/BookForm.tsx

```diff
@@ -0,0 +1,318 @@
+
+interface BookFormProps {
+  bookId?: number
+  hideModal: () => void
+}
+
+interface AuthorOption {
+  label: string
+  value: number
+}
+
+const BookForm: FC<BookFormProps> = ({ bookId, hideModal }) => {
+  const [title, setTitle] = useState('')
+  const [description, setDescription] = useState('')
+  const [authors, setAuthors] = useState<AuthorOption[]>([])
+  const [authorOptions, setAuthorOptions] = useState<AuthorOption[]>([])
+  const [isbn, setIsbn] = useState('')
+  const [quantity, setQuantity] = useState<number | undefined>(1)
+  const [cover, setCover] = useState<File | null>(null)
+  const [coverImage, setCoverImage] = useState<string | undefined>('')
+  const [publishDate, setPublishDate] = useState<Date | null>(null)
+  const [authorFormVisible, setAuthorFormVisible] = useState(false)
+  const [authorFirstName, setAuthorFirstName] = useState('')
+  const [authorLastName, setAuthorLastName] = useState('')
+  const hiddenFileInput = useRef<HTMLInputElement>(null)
+
+  const queryClient = useQueryClient()
+
+  useEffect(() => {
+    if (bookId) {
+      getBook(bookId).then(({ data }) => {
+        setTitle(data.Title)
+        setDescription(data.Description)
+        setAuthors(
+          data.Authors.map((author) => ({
+            label: `${author.Firstname} ${author.Lastname}`,
+            value: author.Id,
+          })),
+        )
+        setIsbn(data.ISBN)
+        setQuantity(data.Quantity)
+        setPublishDate(data.PublishDate)
+        if (data.Cover) {
+          setCoverImage(`data:image/png;base64,${data.Cover}`)
+        }
+      })
+    }
+
+    getAllAuthors().then(({ data }) => {
+      setAuthorOptions(
+        data.map((author) => ({
+          label: `${author.FirstName} ${author.LastName}`,
+          value: author.Id,
+        })),
+      )
+    })
+  }, [])
+
+  const createNewAuthor = async () => {
+    const author: Author = {
+      Id: 0,
+      FirstName: authorFirstName,
+      LastName: authorLastName,
+    }
+
+    await createAuthor(author).catch((error) => {
+      toast.error('Error creating author')
+      throw new Error(error)
+    })
+
+    toast.success('Author created successfully!')
+
+    const { data } = await getAllAuthors().catch((error) => {
+      toast.error('Error retrieving authors')
+      throw new Error(error)
+    })
+
+    setAuthorOptions(
+      data.map((author) => ({
+        label: `${author.FirstName} ${author.LastName}`,
+        value: author.Id,
+      })),
+    )
+
+    setAuthorFormVisible(false)
+    setAuthorFirstName('')
+    setAuthorLastName('')
+  }
+
+  const handleBookFormSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault()
+    if (bookId) {
+      await updateSelectedBook()
+      return
+    }
+    await createNewBook()
+  }
+
+  const updateSelectedBook = async () => {
+    const formData = new FormData()
+    formData.append('Id', bookId?.toString() || '')
```

Comment:
Why are we passing empty string as id? Could this override our api id?

Reply:
The updateSelectBook function is called only when the bookId is defined, so actually it will never be undefined in the function